pkg/sql/colexec/agg/distagg.go

```diff
@@ -429,11 +429,10 @@ func (a *UnaryDistAgg[T1, T2]) UnmarshalBinary(data []byte) error {
 	setDistAggValues[T1, T2](a, a.otyp)
 	a.srcs = decode.Srcs
 	a.maps = make([]*hashmap.StrHashMap, len(a.srcs))
-	m := mpool.MustNewZeroNoFixed()
+	m := mpool.MustNewZeroNoFixed() // this is a hack, will remove later
```

Comment:
This code is so confusing.    Why do you need copyData?   The you copy data into data in line 427.
This mpool need to be deleted.   In this case, why not put it in the operator?   Finally why should we use Marshal/Unmarshal interface ...   Is a operator or operator state?